src/gui/processing/models/qgsmodelgraphicitem.cpp

```diff
@@ -197,4 +221,73 @@ void QgsModelDesignerSocketGraphicItem::paint( QPainter *painter, const QStyleOp
 #endif
 }
 
+
+QColor QgsModelDesignerSocketGraphicItem::socketColor() const
+{
+  return mComponentItem->linkColor( mEdge, mIndex );
+}
+
+
+bool QgsModelDesignerSocketGraphicItem::isDefaultParameterValue() const
+{
+  if ( !mComponent )
+  {
+    return false;
+  }
+
+  const QgsProcessingModelChildAlgorithm *child = dynamic_cast<const QgsProcessingModelChildAlgorithm *>( mComponent );
+
+  if ( !child )
+  {
+    return false;
+  }
+
+  bool isDefaultValue = true;
+
+  // We can only know if the socket should be filled if the algorithm is non null
+  if ( child->algorithm() )
+  {
+    switch ( mEdge )
+    {
+      // Input params
+      case Qt::TopEdge:
+      {
+        QgsProcessingParameterDefinitions params = child->algorithm()->parameterDefinitions();
+
+        if ( mIndex > ( params.length() - 1 ) )
+        {
+          break;
+        }
+
+        const QgsProcessingParameterDefinition *param = params.at( mIndex );
```

Comment:
We can simplify this a little:
```suggestion
        const QgsProcessingParameterDefinitions params = child->algorithm()->parameterDefinitions();
        const QgsProcessingParameterDefinition *param = params.value( mIndex );
        if ( !param )
          break;
```